firmware/projects/FrontController/main.cc

```diff
@@ -0,0 +1,95 @@
+/// @author Blake Freer
+/// @date 2024-02-24
+
+#include "app.h"
+#include "shared/periph/adc.h"
+#include "shared/periph/gpio.h"
+#include "simulink.h"
+
+namespace bindings {
+extern shared::periph::DigitalOutput& driver_speaker;
+extern shared::periph::DigitalOutput& brake_light;
+extern shared::periph::ADCInput& accel_pedal_1;
+extern shared::periph::ADCInput& accel_pedal_2;
+extern shared::periph::ADCInput& brake_pedal;
+extern shared::periph::ADCInput& steering_wheel;
+extern shared::periph::DigitalInput& driver_button;
+
+extern void Initialize();
+}  // namespace bindings
+
+/***************************************************************
+    Objects
+***************************************************************/
+
+Speaker driver_speaker{bindings::driver_speaker};
+BrakeLight brake_light{bindings::brake_light};
+
+Pedal accel_pedal_1{bindings::accel_pedal_1};
+Pedal accel_pedal_2{bindings::accel_pedal_2};
+Pedal brake_pedal{bindings::brake_pedal};
+SteeringWheel steering_wheel{bindings::steering_wheel};
+Button driver_button{bindings::driver_button};
+
+AMKMotor motor_left{};
+AMKMotor motor_right{};
+
+/***************************************************************
+    Globals
+***************************************************************/
+
+SimulinkInput simulink_buffer_input;
+SimulinkOutput simulink_buffer_output;
+
+/***************************************************************
+    Task Functions
+***************************************************************/
+
+/**
+ * @brief Replaces readFromAdc
+ */
+void TakeInput_Task() {
+    simulink_buffer_input.DI_b_DriverButton = driver_button.Read();
+    simulink_buffer_input.DI_V_AccelPedalPos1 = accel_pedal_1.Update();
+    simulink_buffer_input.DI_V_AccelPedalPos2 = accel_pedal_2.Update();
+    simulink_buffer_input.DI_V_BrakePedalPos = brake_pedal.Update();
+    simulink_buffer_input.DI_V_SteeringAngle = steering_wheel.Update();
+}
+
+/**
+ * @brief Copy values to the simulink input struct.
+ */
+void SetControlSystemInputs_Task() {
+    simulink_buffer_output = ControlSystem::Update(simulink_buffer_input);
+}
+
+void ControllerAutogenStep_Task() {}
```

Comment:
Why is this emtpy?

Reply:
With the `output = Update(input)` setup, it doesn't make sense to have separate SetInput(), Step(), and GetOutput() tasks since they all happen in one. This function can be removed

Reply:
Yea lets remove it in that case

Reply:
```suggestion
```